tests/plugins/test_config.py

```diff
@@ -62,6 +62,26 @@ def __init__(self, demo2, simple_int: str, experiment_config):
 
 class RegistryTest(unittest.TestCase):
 
+    def test_recursive_definition(self):
```

Comment:
should we also test 3 levels of nesting? i think we should also test failure cases, where we properly catch cases when the nested object is misconfigured and produce a nice error message.